### PR TITLE
fix: restore Ledger keyring after restoring the vault

### LINKED ISSUE #5721: [Bug] Ledger Keyring is missing after resetting password
**Describe the bug**

**To Reproduce**
_Steps to reproduce the behavior_
1. Go to `Settings`
2. Tap on `Security & Privacy`
3. Scroll down to `Password` and tap on `Change Password`
4. Enter your current password and tap on `Confirm`
5.  Enter a new password and tap on `Reset password`

The ledger account will then be lost and require restarting the connect hardware flow to re-add the account. 

**Expected behavior**

The ledger account remains after the reset password. 

**Smartphone (please complete the following information):**
 - Device: Samsung A32
 - OS: Version 13
 - App Version v5.14.0(1055)

-------------------------------------------------------------
_to be added after bug submission by internal support / PM_
**Severity**

- How critical is the impact of this bug on a user?
- Add stats if available on % of customers impacted
- Is this visible to all users?
- Is this tech debt?

### PR DESCRIPTION
**Pull request by dawnseeker8**
## **Description**

This PR will fix #5721 issue and #8713,  the fix contain following change.
1.  in `Vault.js`,  get the old ledger keyring first before `KeyringController.createNewVaultAndRestore`.
2. Modify existing `restoreLedgerKeyring` method in `Vault.js` to regenerate the ledger keyring with existing account.
3. Unit tests fix for above changes.

## **Related issues**

Fixes: #5721 
          #8713 

## **Manual testing steps**
For #5721 
1. Pair a ledger account with metamask,
2. Go to Settings
3. Tap on Security & Privacy
4. Scroll down to Password and tap on Change Password
5. Enter your current password and tap on Confirm
6. Enter a new password and tap on Reset password
7. After new password update successfully, move to home page and you can see `ledger` account is still there.

For #8713 
1. install the fresh metamask mobile in android. (must be fresh MMM app)
2. restore a account using SRP.
3. when account appear. click the account, and bring up the account drop down list.
4. select Add account and hardware wallet -> Add hardware wallet
5. Select ledger and then allow all permission required.
6. and then add a ledger account in MMM.
7. Make sure `ledger` account is in the account list.
8. Now you can close MMM and relaunch metamask. (or wait for screen lock and reactive phone which will force user relogin to metamask
9. After user login again. you should still see the `ledger` account in the account list.

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/7315988/a5edeabc-e943-4509-996c-b460ccc751b4

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've clearly explained what problem this PR is solving and how it is solved.
- [x] I've linked related issues
- [x] I've included manual testing steps
- [x] I've included screenshots/recordings if applicable
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [x] I’ve properly set the pull request status:
  - [x] In case it's not yet "ready for review", I've set it to "draft".
  - [x] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [x] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [x] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by dawnseeker8** [comment]
> ## [https://bitrise.io/](https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/444d8dbb-7c6a-4140-abbd-c3f8b7569f52) **Bitrise**
> ✅✅✅ `pr_smoke_e2e_pipeline` passed on Bitrise! ✅✅✅
> 
> Commit hash: [4c0ab2d](https://github.com/MetaMask/metamask-mobile/commit/4c0ab2d4aeba93ee541952bcf9ad970ce2be47e9) Build link: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/444d8dbb-7c6a-4140-abbd-c3f8b7569f52
> 
> Note
> 
> * You can kick off another `pr_smoke_e2e_pipeline` on Bitrise by removing and re-applying the `Run Smoke E2E` label on the pull request

Britrise e2e tests pass even `Check Bitrise E2E status` github action failed, MM team will fix github action issue in `main`

**Review by tommasini — COMMENTED** [review]
It seems to follow the same pattern as the `restoreQRKeyring`, but I don't feel 100% comfortable with my knowledge on what's happening on the vault, It would be cool if this have the attention of the accounts team

**Comment by dawnseeker8** [comment]
> It seems to follow the same pattern as the `restoreQRKeyring`, but I don't feel 100% comfortable with my knowledge on what's happening on the vault, It would be cool if this have the attention of the accounts team

Just attached `team-accounts` to ticket so that they can provide the review.

File: app/core/Vault.js [hunk 1]
```diff
@@ -96,6 +100,7 @@ export const recreateVaultWithNewPassword = async (
   const hdKeyring = KeyringController.state.keyrings[0];
   const existingAccountCount = hdKeyring.accounts.length;
 
+  const oldLedgerKeyring = await getLedgerKeyring();
```
Comment (gantunesr) [inline app/core/Vault.js:103]: nit
```suggestion
  const ledgerKeyring = await getLedgerKeyring();
```

**Review by gantunesr — DISMISSED** [review]
Just a minor comment

**Review by gantunesr — APPROVED** [review]

**Comment by dawnseeker8** [comment]
E2e tests pass even `Check Bitrise E2E status` failed 
Here is the result, ready to be merged.
https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/1cfc64ba-9131-4ecb-9e40-2ac194857e01

**Pull request merged**